Rafael-G-C/pyDFTD3
# Connect testing to new code

Summary of what happens in this PR:
- Tests now only test the new code (against old reference numbers so functionality is preserved)
- Test time got shorter
- Added the Becke-Johnson damping also to the new code
- Renamed `dftd3` to `dftd3_old`
- Did not remove any code in `dftd3_old`
- There is no command line interface in the new code (see follow-up below)

Follow-up steps for later:
- Decide whether to re-introduce a command line interface in front of the library
- Update README
- Maybe disconnect fork-relation to avoid sending PRs accidentally to upstream
- Remove `examples` since they are not relevant for this code?
- Maybe introduce wrapper functions to simplify the calls when differentiating
